### DISCUSSION TITLE
Is the method call from JavaScript to C# in CefSharp single-threaded?

### DISCUSSION THREAD

**Discussion by bbbrowser1** [cefsharp/CefSharp#4802] [Q&A] [answered]
Front-end code for calling the browser method:
```
await CefSharp.BindObjectAsync("asyncZBrowser");
asyncZBrowser.execute(JSON.stringify(options)).then(function (res) {
    hander && hander(JSON.parse(res));
});
```
Browser method:
```
public class AsyncZBrowser
{
    // Main entry point for the call
    public string execute(string paramStr)
    {
        Console.WriteLine("【Thread ID】" + System.Threading.Thread.CurrentThread.ManagedThreadId + ':' + paramStr);
        return new Execute().execute(paramStr);
    }
}
```
Why, when making multiple calls to asyncZBrowser.execute simultaneously on the front-end Vue page, do the requests come in sequentially, and the thread IDs printed out for multiple requests are the same?

**Comment by amaitland** [comment, marked as answer]
> Why, when making multiple calls to asyncZBrowser.execute simultaneously on the front-end Vue page, do the requests come in sequentially, and the thread IDs printed out for multiple requests are the same?

By default method calls are processed sequentially. You can switch to concurrent execution (on the thread pool) by setting the following before you create `ChromiumWebBrowser` instances.

```c#
CefSharpSettings.ConcurrentTaskExecution = true;
```
